src/main/scala/org/scalacheck/Gen.scala

```diff
@@ -649,15 +649,27 @@ object Gen extends GenArities{
   def identifier: Gen[String] = (for {
     c <- alphaLowerChar
     cs <- listOf(alphaNumChar)
-  } yield (c::cs).mkString).suchThat(_.forall(c => c.isLetter || c.isDigit))
+  } yield (c::cs).mkString).suchThat(_.forall(_.isLetterOrDigit))
+
+  /** Generates a string of digits */
+  def numStr: Gen[String] =
+    listOf(numChar).map(_.mkString).suchThat(_.forall(_.isDigit))
```

Comment:
I think we should avoid `suchThat` https://github.com/rickynils/scalacheck/issues/256#issuecomment-244584016